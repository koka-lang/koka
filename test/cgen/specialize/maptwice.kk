inline fun maptwice(f : int -> _) : total list<list<int>> {
  [[1,2,3]].map(fn(xs) xs.map(f))
}

fun main() {
  maptwice(inc)[0].unjust[0].unjust.println
}