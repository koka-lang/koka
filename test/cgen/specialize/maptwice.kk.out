2
add default effect for std/core/exn
 
cgen/specialize/maptwice/.hmain: () -> console ()
cgen/specialize/maptwice/.lift000-maptwice: (f : (int) -> int, xs : list<list<int>>) -> total list<list<int>>
cgen/specialize/maptwice/.lift000-main: (f : (int) -> int, xs : list<list<int>>) -> total list<list<int>>
cgen/specialize/maptwice/.mlift000-main: (int) -> exn ()
cgen/specialize/maptwice/.mlift000-main: (list<int>) -> exn ()
cgen/specialize/maptwice/main: () -> <console,exn> ()
cgen/specialize/maptwice/maptwice: (f : (int) -> int) -> total list<list<int>>